alacritty/src/renderer/platform.rs

```diff
@@ -20,23 +22,36 @@ use winit::platform::x11;
 /// Create the GL display.
 pub fn create_gl_display(
     raw_display_handle: RawDisplayHandle,
-    _raw_window_handle: Option<RawWindowHandle>,
+    display_api: DisplayApiPreference,
 ) -> GlutinResult<Display> {
+    let display = unsafe { Display::new(raw_display_handle, display_api)? };
+    log::info!("Using {}", { display.version_string() });
+    Ok(display)
+}
+
+pub fn preferred_display_api(_raw_window_handle: Option<RawWindowHandle>) -> DisplayApiPreference {
     #[cfg(target_os = "macos")]
-    let preference = DisplayApiPreference::Cgl;
+    return DisplayApiPreference::Cgl;
 
     #[cfg(windows)]
-    let preference = DisplayApiPreference::Wgl(Some(_raw_window_handle.unwrap()));
+    return DisplayApiPreference::Wgl(Some(_raw_window_handle.unwrap()));
 
     #[cfg(all(feature = "x11", not(any(target_os = "macos", windows))))]
-    let preference = DisplayApiPreference::GlxThenEgl(Box::new(x11::register_xlib_error_hook));
+    {
+        let egl_over_glx = match env::var("EGL_OVER_GLX") {
+            Ok(egl) => egl,
+            Err(_) => panic!("Could not find environment variable 'EGL_OVER_GLX'"),
+        };
 
-    #[cfg(all(not(feature = "x11"), not(any(target_os = "macos", windows))))]
-    let preference = DisplayApiPreference::Egl;
+        if matches!(egl_over_glx.as_ref(), "yes" | "true" | "1") {
+            DisplayApiPreference::EglThenGlx(Box::new(x11::register_xlib_error_hook))
+        } else {
+            DisplayApiPreference::GlxThenEgl(Box::new(x11::register_xlib_error_hook))
+        }
+    }
 
-    let display = unsafe { Display::new(raw_display_handle, preference)? };
-    log::info!("Using {}", { display.version_string() });
-    Ok(display)
+    #[cfg(all(not(feature = "x11"), not(any(target_os = "macos", windows))))]
+    return DisplayApiPreference::Egl;
 }
```

Comment:
Could you just merge both functions and not pass `DisplayApi` to the function? there's no need to that, you can pick the display based on the env variable just fine.